### ISSUE TITLE
Multiple stacked "do you want to save" boxes blocking each other

### ISSUE THREAD

**Issue by m1ga** [pulsar-edit/pulsar#304]
### Thanks in advance for your bug report!

- [X] Have you reproduced this issue in incognito/private browsing?
- [X] Have you made sure you issue doesn't already exist?

### Where is the URL that this occurs?

-

### What's your issue?

When closing an editor with an unsaved text tab it will show 4 "do you want to save them" alerts when opening the editor again:
![Screenshot_20230105_174759](https://user-images.githubusercontent.com/4334997/210835162-9bc9974b-732c-4f41-9738-e43dbaccc35f.png)

those are normally stacked on top and sometimes in the wrong order so you can click the visible item to cancel it. You'll have to drag the windows away to find the latest one (that might be a KDE issue). But it should only show one alert!

### Which OS/Browser/Version does this happen on?

Fedora Linux 37
Pulsar  : 1.100.2022122507
Electron: 12.2.3
Chrome  : 89.0.4389.128
Node    : 14.16.0

### Steps to Reproduce/Additional Details:

* start Pulsar in a folder
* ctrl + n for a new file - write some text (don't save it)
* close pulsar
* open it again
* you should see multiple "save" alerts

**Comment by bobemoe** [comment]
I suspect this is due to they way Pulsar handles restoring unsaved tabs. It can be a bit confusing as there are 4 different ways this is handled, see https://github.com/pulsar-edit/pulsar/issues/281#issuecomment-1374899805 for a summary. I think this "logic" is probably up for review!

As you say you "start Pulsar in a folder" I guess this means you have a project folder open. The way untitled tabs should be handled in this case is silently saved and restored.

I do occasionally (usually when computer being slow)  see a "do you want to save" dialogue while I am closing Pulsar in those conditions but the dialogue is just a flash and disappears at the same time the main window closes. It does not reappear upon opening Pulsar but the tabs have been restored.

This leads me to believe Pulsar always shows that dialogue but closes it as soon as the tab is automatically saved. I think you may be right that KDE is interfering somehow as I expect programmatically closing that window is dependant on the Window Manager.  I am using Xfce4 here, which may explain why I can't recreate the issue.

Sorry I have no solutions but hopefully I might have provided some clues :) I'll see if I can find the related code.

**Comment by m1ga** [comment]
Thanks for the input.
There was an old Atom bug https://github.com/atom/atom/issues/17459 with almost the same behavior. It wasn't fully fixed as you can see in the last comment in that thread.

"start Pulsar in a folder": I always use the CLI to start my editors. So I run `pulsar .` in the folder I want to open. Not sure if that is a "project folder" at the end. I don't open multiple projects in one editor. When I run `pulsar .` in another folder it will open another instance of Pulsar

**Comment by bobemoe** [comment]
Just tested on clean Pulsar install and `pulsar .` does indeed open the current directory as a project folder.

With that project folder open, for me multiple untitled tabs are saved and restored silently.

Opening with just `pulsar` opens with no project folder (must be clean install or previous project folder is opened) and in this case closing with multiple untitled tabs does create multiple stacked save dialogues. However in my case they are stacked in the correct order to be able to click through without having to un-shuffle them.

Some interesting comments in these commits, but they are pretty old so not sure how relevant.  https://github.com/pulsar-edit/pulsar/commit/f0448ba231c55be7cdb82cbdb20b864c4ece6db4, https://github.com/pulsar-edit/pulsar/commit/0f6eadcfceb27784a65f9898b301fa5839c71210 Gives some idea of what code is involved though, but I'm afraid I don't understand it enough to be able to dig much deeper.

**Comment by bobemoe** [comment]
In you original report it sounds like you only have 1 untitled tab, but see 4 dialogues. Is that correct?

Also, could you confirm if this is also repeatable on the final build of Atom?

**Comment by m1ga** [comment]
> In you original report it sounds like you only have 1 untitled tab, but see 4 dialogues. Is that correct?
> 
> Also, could you confirm if this is also repeatable on the final build of Atom?

yes to both

**Comment by savetheclocktower** [comment]
@m1ga, does this still happen to you?

**Comment by m1ga** [comment]
no, I don't see it currently happening anymore. It will just close without asking to save now and reopen the unsaved tabs as they were before. So I guess we can close this one now as the behavior changed
